### ISSUE TITLE
CentOS 8.4: Copying files from host to guest via Drag&Drop leads to incomplete copies

### ISSUE THREAD

**Issue by rgoadvice** [vmware/open-vm-tools#552]
Working 10 years with VMware, have to state that Copy&Paste is probably one of the most unrealiable features.  Took me quite a while to find out that Wayland is not supported yet. No warnings, no nothing. I switched back to X11, however, when I drag&drop larger files (about 100MB) from my host (Windows 10 19043.1288) to CentOS 8 (8.4.2105, open-vm-tools-desktop-11.2.0-2.el8.x86_64), they are not completely copied, instead just a few mega bytes are copied, sometimes more, sometimes less. Also, sometimes the VMWare Workstation Popup occurs, that the file is copied, but nothing happens afterwards. The file is not moved to the chosen target folder. Note that in any of the cases above the file is completly copied to "~/.cache/vmware" however then either not moved or incompletly moved to the chosen target directory.

Laptop uses very fast NVME SSD. Running on latest VMWare Workstation 16.2.0 build-18760230.

**Comment by lousybrit** [comment]
Thanks for reporting this issue. Thanks for including all the version information.
Host Windows 10 19043.1288
Workstation 16.2.0 build-18760230
VM CentOS 8 (8.4.2105, open-vm-tools-desktop-11.2.0-2.el8.x86_64)

I will file a PR for our Drag and Drop engineers to look at this issue.

From your description, the files are copied to the staging directory in their entirety but then not moved to the target location correctly.
The direction of the copy of larger files >= 100BM or thereabouts, the file copy is from the host to the guest.

We will report back here with updates on reproduction and any temporary workarounds are possible.

Steve

**Comment by rgoadvice** [comment]
Yes, that is indeed the issue: The file is completely copied to "~/.cache/vmware" but either not moved at all into the chosen target directory or copied incompletely. Looks like the movement to the target directory happens too early (before the file was successfully copied to "~/.cache/vmware). Here you will find a video demonstrating the issue:

https://user-images.githubusercontent.com/6525437/140866600-f078f710-db71-4086-9051-33df2e7067ca.mp4

Bye
Ralf
